src/S3/StreamWrapper.php

```diff
@@ -830,10 +839,12 @@ private function createSubfolder($path, array $params)
         $params['Body'] = '';
 
         // Fail if this pseudo directory key already exists
-        if ($this->getClient()->doesObjectExist(
+        $method = self::$useV2Existence ? 'doesObjectExistV2' : 'doesObjectExist';
```

Comment:
Since each of these are duplicated once, you could have a method below that is checkObjectExists() and that makes the two calls below; not a huge deal

```
$this->checkObjectExist($params['Bucket'], $params['Key']);

private function checkObjectExists($bucket, $key) {
      $method = self::$useV2Existence ? 'doesObjectExistV2' : 'doesObjectExist';
      $this->getClient()->method($bucket, $key);
}
```

Something like that.  Just so modifying it in the future is easier.